src/shaders/_prelude.vertex.glsl

```diff
@@ -48,12 +48,18 @@ float unpack_mix_vec2(const vec2 packedValue, const float t) {
 }
 
 // Unpack a pair of paint values and interpolate between them.
-vec4 unpack_mix_vec4(const vec4 packedColors, const float t) {
+vec4 unpack_mix_color(const vec4 packedColors, const float t) {
     vec4 minColor = decode_color(vec2(packedColors[0], packedColors[1]));
     vec4 maxColor = decode_color(vec2(packedColors[2], packedColors[3]));
     return mix(minColor, maxColor, t);
 }
 
+vec4 unpack_mix_vec4(const vec4 patternVertices, const float t) {
+    // line pattern location vertices are not packed, and they use a different t
+    // for interpolation that is specific to cross-faded properties
+    return patternVertices;
+}
```

Comment:
I agree it is confusing, but since cross faded properties are more the exception than the rule, I don't know if adding a new slew of `#define`s will be clearer in the long run I took an alternate approach – let me know if this works for you!